be/src/storage/chunk_helper.cpp

```diff
@@ -373,4 +373,19 @@ std::shared_ptr<vectorized::Chunk> ChunkHelper::new_chunk(const std::vector<Slot
     return chunk;
 }
 
+void ChunkHelper::reorder_chunk(const TupleDescriptor& tuple_desc, vectorized::Chunk* chunk) {
+    return reorder_chunk(tuple_desc.slots(), chunk);
+}
+
+void ChunkHelper::reorder_chunk(const std::vector<SlotDescriptor*>& slots, vectorized::Chunk* chunk) {
```

Comment:
I think we can do some optimization: that if order is correct, we don't need to do anything at all. 

But we can put this opt into another PR.

Reply:
If so, I have to for-loop twice, first is to check if the order is correct, second is to reorder it if the order is not correct. I'm not sure the benefit of not swapping columns deserves doing this.

Reply:
for-loop is just CPU operation, faster than allocate memory for chunk. And you can make for-loop fast with some efforts.

Reply:
There is few memory allocation for chunk, chunk only holds shared pointer to columns. The typical memory footprint of a `std::shared_ptr` is 44B. Say there are 1k columns in the chunk, the memory footprint of chunk will be ~44KB

Reply:
It's not about memory footprint, it's about memory allocation.

Allocate 1k times is bad. and think in chunk there are map & vector data structures.

Reply:
The memory allocation is not linear, typical memory allocation strategy is twice the space. So assume there are 1k columns, it will only allocate 10 times for `Columns`

Reply:
OK.